### ISSUE TITLE
GLSL preprocessor appends define onto #version value causing exception?

### ISSUE THREAD

**Issue by pizthewiz** [cinder/Cinder#909]
Say I have a passthrough shaders:
```C++
const std::string EffectorNode::sVertexShaderPassThrough = R"(
    #version 150
    uniform mat4 ciModelViewProjection;

    in vec4 ciPosition;
    in vec2	ciTexCoord0;

    out vec2 vTexCoord0;

    void main() {
        vTexCoord0 = ciTexCoord0;
        gl_Position = ciModelViewProjection * ciPosition;
    }
)";
const std::string EffectorNode::sFragmentShaderPassThrough = R"(
    #version 150
    uniform sampler2D image;

    in vec2 vTexCoord0;

    out vec4 oFragColor;

    void main() {
        oFragColor = texture(image, vTexCoord0);
    }
)";
```

if any defines are added:
```C++
auto format = gl::GlslProg::Format().vertex(sVertexShaderPassThrough).fragment(FragmentShaderGaussianBlur);
format.define("ACK", toString(1));
mShader = gl::GlslProg::create(format);
```
the resulting `preprocessedSource` from [`GlslProg::LoadShader`](https://github.com/cinder/Cinder/blob/glNext/src/cinder/gl/GlslProg.cpp#L628) has the define tacked right onto the version number:
```gdb
(lldb) p preprocessedSource
(std::__1::string) $1 = "    #version 150#define ACK 1\n\n\n    uniform mat4 ciModelViewProjection;\n\n    in vec4 ciPosition;\n    in vec2\tciTexCoord0;\n\n    out vec2 vTexCoord0;\n\n    void main() {\n        vTexCoord0 = ciTexCoord0;\n        gl_Position = ciModelViewProjection * ciPosition;\n    }\n"
```
which I think is causing  a compile error:
```
|error  | void cinder::app::AppBase::executeLaunch()[140] Uncaught exception, type: cinder::gl::GlslProgExc, what : VERTEX: ERROR: 0:1: '' : syntax error: #version

libc++abi.dylib: terminating with uncaught exception of type cinder::gl::GlslProgExc: VERTEX: ERROR: 0:1: '' : syntax error: #version
```

This seems to be a regression in glNext over the last week or so, #895 seems related but I think the code path that was changed there is when the preprocessor inserts a version, so perhaps not.

**Comment by pizthewiz** [comment]
Oh actually no, even the title of #895 indicates it is for _user_-provided versions so it could be the cause after all.

**Comment by richardeakin** [comment]
Ah, I think we finally know what the '\n' was added after #version, which I [removed here](#895). So I suppose the correct thing to do is only added that extra newline if we are about to add a define.. I can do this later this evening.

**Comment by richardeakin** [comment]
Fixed with #910
